### ISSUE TITLE
[WebToolsE2E][Aspire][Safari] Opening the "Aspire Dashboard" URL of a deployed Aspire project fails with error "Misdirected Request" on Mac.

### ISSUE THREAD

**Issue by v-sherryfan** [dotnet/aspire#4854]
**INSTALL STEPS**

1. Clean machine: MacOS M1 (Sonoma 14.5)
2. Install latest released SDK 8.0.303
3. Install latest released Aspire 8.0.2
4. Install latest released azd 1.9.5

**REPRO STEPS**

1. Open Terminal, run following command to create an Aspire Starter App project and deploy it 
```
dotnet new aspire-starter -o aspire1
cd aspire1/aspire1.AppHost
azd init
azd up
```
2. After successful deployment, open the deployed "Aspire Dashboard" URL

**NOTE:**

- This issue does not repro on Edge browser, only on Safari.
- This issue also reproduce on Aspire 8.1.

**ACTUAL**
Opening the deployed 'Aspire Dashboard' URL displays a "Misdirected Request" error. However, sometimes after waiting for a while and manually refreshing the page, it can be displayed normally.
![image](https://github.com/dotnet/aspire/assets/71361488/e2e6dc9d-99d1-4d1d-871d-f448696896f1)

**EXPECTED**
![image](https://github.com/dotnet/aspire/assets/71361488/4b43aac3-bbb3-43c2-b711-7ab73af64512)

**Comment by leslierichardson95** [comment]
@snehapar9 Is this potentially an ACA issue?

**Comment by snehapar9** [comment]
Thanks for opening this issue @v-sherryfan! Does this consistently repro on Safari? I was able to successfully access the Aspire dashboard on Safari. This may be a transient issue.

**Comment by v-sherryfan** [comment]
@snehapar9 Verified the issue using aspire 8.1.1 on Safari, now the issue does not reproduce when clicking on deployed aspire dashboard links, but does consistently reproduce on deployed webfrontend links.
![image](https://github.com/user-attachments/assets/bf5bf520-0780-472e-9d95-612b811c20bb)
![image](https://github.com/user-attachments/assets/7c88ebad-f377-4544-b92e-e34c47dfb43f)

**Comment by adamint** [comment]
@snehapar9

**Comment by v-sherryfan** [comment]
@davidfowl Yes, it still reproduced using SDK 9.0.102 + Aspire 9.1.0 on MacOS arm64.

**Comment by davidfowl** [comment]
Do you still see this?

**Comment by v-reinawang** [comment]
Not repro using SDK 9.0.301 + Aspire 9.3.1 on MacOS arm64.
